src/plugins/controls/public/options_list/components/options_list_popover_action_bar.tsx

```diff
@@ -69,24 +65,32 @@ export const OptionsListPopoverActionBar = ({
               onChange={(event) => updateSearchString(event.target.value)}
               value={searchString.value}
               data-test-subj="optionsList-control-search-input"
-              placeholder={
-                totalCardinality
-                  ? OptionsListStrings.popover.getTotalCardinalityPlaceholder(totalCardinality)
-                  : undefined
-              }
+              placeholder={OptionsListStrings.popover.getSearchPlaceholder()}
               autoFocus={true}
             />
           </EuiFlexItem>
+          {allowExpensiveQueries && (
+            <EuiFlexItem grow={false}>
+              <EuiToolTip
+                content={OptionsListStrings.popover.getCardinalityTooltip(totalCardinality)}
+              >
+                <EuiBadge
+                  data-test-subj="optionsList-cardinality-badge"
+                  className="optionsList__cardinalityBadge"
+                >
+                  {totalCardinality.toLocaleString()}
+                </EuiBadge>
+              </EuiToolTip>
+            </EuiFlexItem>
+          )}
```

Comment:
This is how the new cardinality badge interacts with the old "invalid selections" badge:

![image](https://user-images.githubusercontent.com/8698078/215585580-f945a802-80ba-4ad5-95c4-bb8c58756e3a.png)